internal/controllers/addon/phase_ensure_monitoring_federation.go

```diff
@@ -123,15 +123,15 @@ func (r *AddonReconciler) reconcileServiceMonitor(
 		return fmt.Errorf("getting ServiceMonitor: %w", err)
 	}
 
-	if len(currentServiceMonitor.OwnerReferences) == 0 ||
-		!controllers.HasEqualControllerReference(currentServiceMonitor, serviceMonitor) {
+	ownedByAddon := controllers.HasEqualControllerReference(currentServiceMonitor, serviceMonitor)
+	specChanged := !equality.Semantic.DeepEqual(currentServiceMonitor.Spec, serviceMonitor.Spec)
+	if !ownedByAddon {
 		// TODO: remove this condition once resourceAdoptionStrategy is discontinued
-		if strategy != addonsv1alpha1.ResourceAdoptionAdoptAll {
+		if strategy != addonsv1alpha1.ResourceAdoptionAdoptAll && !ownedByAddon {
```

Comment:
You can remove the outer `if` here since nested `if`s imply an `AND` operation.
I would also move the `specChanged` declaration to the next block where it is used, but that's a nit.

Reply:
Ohh, a silly one, my bad! On it.